### PR TITLE
character-set-and-collation: support use `LIKE` to build range for new collation columns

### PR DESCRIPTION
**Pull request by time-and-fate**
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/pull/51164

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by songrijie — APPROVED** [review]
LGTM

File: character-set-and-collation.md [hunk 1]
```diff
@@ -111,7 +111,7 @@ SHOW COLLATION;
 >
 > TiDB 会错误地将 `latin1` 视为 `utf8` 的子集。当用户存储不同于 `latin1` 和 `utf8` 编码的字符时，可能会导致意外情况出现。因此强烈建议使用 `utf8mb4` 字符集。详情参阅 [TiDB #18955](https://github.com/pingcap/tidb/issues/18955)。
 >
-> 如果查询条件中包含对字符串前缀的 `LIKE` 过滤， 比如 `LIKE 'prefix%'`，并且该列被设置为非二进制排序规则（即排序规则的后缀不是以 `_bin` 结尾），那么 TiDB 优化器暂时无法把这个过滤条件转化为范围扫描 (Range Scan)，而是用全量扫描代替。因此这类 SQL 有可能造成超出预期的资源消耗。
+> 在 TiDB v7.5.0 中，如果查询条件中包含对字符串前缀的 `LIKE` 过滤， 比如 `LIKE 'prefix%'`，并且该列被设置为非二进制排序规则（即排序规则的后缀不是以 `_bin` 结尾），那么 TiDB 优化器暂时无法把这个过滤条件转化为范围扫描 (Range Scan)，而是用全量扫描代替。因此这类 SQL 有可能造成超出预期的资源消耗。从 v7.5.1 开始，TiDB 优化器不再有这个限制。
```
Comment (Oreoxmt) [inline character-set-and-collation.md:114]:
```suggestion
> 在 TiDB v7.5.0 中，如果查询条件中包含对字符串前缀的 `LIKE` 过滤， 比如 `LIKE 'prefix%'`，并且该列被设置为非二进制排序规则（即排序规则的后缀不是以 `_bin` 结尾），那么 TiDB 优化器暂时无法把这个过滤条件转化为范围扫描 (Range Scan)，而是用全量扫描代替。因此这类 SQL 有可能造成超出预期的资源消耗。从 v7.5.1 开始，TiDB 优化器移除该限制。
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

File: character-set-gbk.md [hunk 2]
```diff
@@ -88,7 +88,7 @@ SHOW COLLATION WHERE CHARSET = 'gbk';
 
 * 对于 `ENUM` 和 `SET` 类型中的二进制字符，TiDB 目前都会将其作为 `utf8mb4` 字符集处理。
 
-* 如果查询条件中包含对字符串前缀的 `LIKE` 过滤，比如 `LIKE 'prefix%'`，并且该列被设置为 GBK 的排序规则（`gbk_bin` 或 `gbk_chinese_ci`），那么 TiDB 优化器暂时无法把这个过滤条件转化为范围扫描 (Range Scan)，而是用全量扫描代替。因此这类 SQL 有可能造成超出预期的资源消耗。
+* 在 TiDB v7.5.0 中，如果查询条件中包含对字符串前缀的 `LIKE` 过滤，比如 `LIKE 'prefix%'`，并且该列被设置为 GBK 的排序规则（`gbk_bin` 或 `gbk_chinese_ci`），那么 TiDB 优化器暂时无法把这个过滤条件转化为范围扫描 (Range Scan)，而是用全量扫描代替。因此这类 SQL 有可能造成超出预期的资源消耗。从 v7.5.1 开始，TiDB 优化器不再有这个限制。
```
Comment (Oreoxmt) [inline character-set-gbk.md:91]:
```suggestion
* 在 TiDB v7.5.0 中，如果查询条件中包含对字符串前缀的 `LIKE` 过滤，比如 `LIKE 'prefix%'`，并且该列被设置为 GBK 的排序规则（`gbk_bin` 或 `gbk_chinese_ci`），那么 TiDB 优化器暂时无法把这个过滤条件转化为范围扫描 (Range Scan)，而是用全量扫描代替。因此这类 SQL 有可能造成超出预期的资源消耗。从 v7.5.1 开始，TiDB 优化器移除该限制。
```

**Review by hfxsd — APPROVED** [review]

**Pull request merged**